### PR TITLE
Auto-add connector docs pages to catalog sidebar and update checklist

### PR DESCRIPTION
**Pull request by sherifnada**
old source connectors not in the new list
[]
new source connectors not in the current list
```
[
  'integrations/sources/chargify',
  'integrations/sources/commercetools',
  'integrations/sources/faker',
  'integrations/sources/hellobaton',
  'integrations/sources/mongodb',
  'integrations/sources/pivotal-tracker',
  'integrations/sources/retently',
  'integrations/sources/rki-covid',
  'integrations/sources/sftp',
  'integrations/sources/talkdesk-explore',
  'integrations/sources/tidb'
]
```
current destination connectors not in the new list
[]
new destination connectors not in the current list
```
[ 'integrations/destinations/aws-datalake' ]
```

### CODE REVIEW

**Comment by supertopher** [comment]
TLDR: let's see if the docs team is interested in adopting some form of auto generator for the entire docs folder.  I think it will make it easier for all of us including OSS to contribute.

this looks fantastic for people adding connector documentation.  This seems like a decent time to have a conversation about the sidebar in general though.  It seems like the problem maybe exists everywhere in the docs folder.  The JSON thing in here has always felt hard to use to me.  I get lost in the nested structure in there.  Alphabetizing JSON items isn't fun.  I can't imagine that's intuitive for new contributors.  Additionally: Your approach I think is actually a bit more intuitive that the auto-generator docusaurus has built in.

namely: the docusaurus auto generator [we use in cloud](https://github.com/airbytehq/airbyte-cloud/blob/master/docusaurus/sidebars.js) requires a readme for a folder to be "seen".  This forces a landing page for the drop down which is nice, but telling new people contributing to docs that isn't very easy.  Your approach is more greedy making the 1:1 look alot easier to see.  There are a couple other just not intuitive things in the cloud version. ie: the first `# heading` in markdown file is also the sidebar title.  

That said I don't want to step on the toes of the docs team.  If it's working well over there for @Amruta-Ranade and the rest of the team your fix seems like a great way to make connector documentation easier for new contributors and we can leave well enough alone.  I think we could help make this easier for both us and open source if we just bite the bullet and make the folder structure look how we want the sidebar to look.  In full disclosure I don't think I really understand how large an ask that is to the docs team or as a task.  The folder [does look relatively organized](https://gist.github.com/supertopher/6fed84c50490f102a00bd9baab11e058) to my untrained eye though.  

I too miss summary.md


and this is a screenshot of your change for the history book 
Screen Shot 2022-06-04 at 9 24 33 AM

**Review by supertopher — APPROVED** [review]
let's make sure we express these changes clearly to the docs team, but I like this especially for connectors and sources

**Comment by Amruta-Ranade** [comment]
+1000 to autogenerating the sidebar. The nested structure of the JSON file does get confusing and at times, frustrating. Happy to help in any way I can.
